eth2/operation_pool/src/attestation.rs

```diff
@@ -40,15 +58,16 @@ impl<'a, T: EthSpec> MaxCover for AttMaxCover<'a, T> {
     fn update_covering_set(
         &mut self,
         best_att: &Attestation<T>,
-        covered_validators: &BitList<T::MaxValidatorsPerCommittee>,
+        covered_validators: &HashMap<u64, u64>,
     ) {
         if self.att.data.slot == best_att.data.slot && self.att.data.index == best_att.data.index {
-            self.fresh_validators.difference_inplace(covered_validators);
+            self.fresh_validators_rewards
+                .retain(|k, _| covered_validators.contains_key(k))
```

Comment:
Should be `!covered_validators.contains_key(k)`, because we want to keep all the validators who aren't covered.

On a related note, to check the correctness of this impl, it might be nice to have some tests? We can open a separate issue for that though.

Reply:
Haha I'm still struggling to make CI pass, not quite done yet but yet indeed it should be `!` ! :)
Maybe I should simply add the tests in this PR?

Reply:
Yeah any tests you can think of that might be useful, feel free to add. Perhaps the ones in `lib.rs` are sufficient (if they were failing without the `!`)